src/auc.py

```diff
@@ -0,0 +1,33 @@
+import json_repair
+import numpy as np
+import pandas as pd
+
+from sklearn.metrics import roc_auc_score
+from sklearn.preprocessing import MinMaxScaler
+
+
+def calculate_auc(group):
+    if group["ground_truth"].nunique() < 2:
+        return np.nan  # AUC-ROC cannot be calculated if there is only one class in the group
+    return roc_auc_score(group["ground_truth"], group["normalized_value"])
+
+
+ground_truth_df = pd.read_csv("../results/ground_truth.csv")
+ground_truth_df["complex"] = [0 if row["Simple/Complex"] == "Simple" else 1 for _, row in ground_truth_df.iterrows()]
+ground_truth_map = {row["Dataset"]: row["complex"] for _, row in ground_truth_df.iterrows()}
+
+with open("../results/logs.jsonl", "r") as f:
+    lines = list(f)
+
+results_df = pd.DataFrame([json_repair.loads(line) for line in lines])
```

Comment:
Might be missing something, but can't you use `pd.read_json("../results/logs.jsonl", lines=True)` to read jsonl?